### PR TITLE
Admin Router: Deduplicate service.lua and related files

### PR DESCRIPTION
**Pull request by vespian**
## High-Level Description

This PR:
* consolidates service.lua code into a single file.
* makes service routing code more efficient by pre-calculating some of the data in cache.lua/during cache fetch stage.
* refactors code to better accommodate incoming hierarchical service names
* cleans up the code, adds some docstrings

## Related Issues
- [DCOS-14589](https://jira.mesosphere.com/browse/DCOS-14589) Admin Router: Dedup of `/service/` endpoint

## Dependencies

Related EE PR: https://github.com/mesosphere/dcos-enterprise/pull/798

### CODE REVIEW

**Comment by vespian** [comment]
@mesosphere-mergebot label Ready for review

**Comment by BenWhitehead** [comment]
@vespian can you please rebase this agains `master` to pick up the pinned python dependency which should make `teamcity/create-release-pr` be able to build.

**Comment by vespian** [comment]
@mesosphere-mergebot label Work in Progress

**Comment by vespian** [comment]
@BenWhitehead This PR is waiting for #1438 to be merged, I have relabeled is as Work in progress for now. After #1438 gets in, I will rebase & cleanup  this one and make sure that tests are green.

Thanks!

File: packages/adminrouter/extra/src/master/agent.lua [hunk 1]
```diff
@@ -6,19 +6,18 @@ if state == nil then
     return ngx.exit(ngx.HTTP_SERVICE_UNAVAILABLE)
 end
 
-for _, agent in ipairs(state["slaves"]) do
-    if agent["id"] == ngx.var.agentid then
-        local split_pid = agent["pid"]:split("@")
-        local host_port = split_pid[2]:split(":")
-        ngx.var.agentaddr = DEFAULT_SCHEME .. host_port[1]
-        ngx.var.agentport = host_port[2]
+agent_pid = state['agent_pids'][ngx.var.agentid]
```
Comment (gpaul) [inline packages/adminrouter/extra/src/master/agent.lua]: nice one

File: packages/adminrouter/extra/src/master/cache.lua [hunk 2]
```diff
@@ -269,18 +269,48 @@ end
 local function fetch_and_store_state_mesos(auth_token)
     -- Fetch state JSON summary from Mesos. If successful, store to SHM cache.
     -- Expected to run within lock context.
-    local mesosRes, err = request(UPSTREAM_MESOS .. "/master/state-summary",
-                                  false,
-                                  auth_token)
+    local response, err = request(UPSTREAM_MESOS .. "/master/state-summary",
+                                           false,
```
Comment (gpaul) [inline packages/adminrouter/extra/src/master/cache.lua]: should this indentation get updated to line up with `UPSTREAM_MESOS`?

Comment (vespian) [inline packages/adminrouter/extra/src/master/cache.lua]: it should, fixing!

**Comment by vespian** [comment]
Closing in favour of https://github.com/mesosphere/dcos/pull/16

**Pull request closed without merging**